### DISCUSSION TITLE
Cli and multiple input files

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#883] [Q&A] [answered]
hi, is there an easy option / example to have a cli that processes a complete source dir/project and not just a single file (similar as xtext-gradle/maven-plugin would do? thanks

**Comment by msujew** [comment]
Hey @cdietrich,

we do have an example in the domainmodel language. See [here](https://github.com/langium/langium/blob/main/examples/domainmodel/src/cli/cli.ts). The `root` argument specifies the directory and [eventually loads](https://github.com/langium/langium/blob/2a895e6b56535318fd24283cfdebc9e6838d52a5/examples/domainmodel/src/cli/cli-util.ts#L46-L58) a whole workspace.

**Reply by cdietrich** [reply]
i saw this. but the generator seems do be called only for one file or did i missread the code https://github.com/langium/langium/blob/2a895e6b56535318fd24283cfdebc9e6838d52a5/examples/domainmodel/src/cli/generator.ts#L22

**Reply by msujew** [reply]
Right. You can always access all documents through the `services.shared.workspace.LangiumDocuments.all` property if you want to iterate over all of them. They should be all available after `initializeWorkspace` has been awaited.

**Reply by cdietrich** [reply]
ok thx. will have a look

**Reply by cdietrich** [reply]
it actually works as expected

**Comment by cdietrich** [comment, marked as answer]
To answer this myself

```
        services.shared.workspace.LangiumDocuments.all.forEach(d => {
            const domainmodel = d.parseResult.value as Domainmodel
            const generatedDirPath = generateJava(domainmodel, d.uri.toString(), opts.destination);
            if (!opts.quiet) {
                console.log(chalk.green(`Java classes generated successfully: ${chalk.yellow(generatedDirPath)}`));
            }
        })
```
worked fine
